diff --git a/public/css/style.css b/public/css/style.css
index 89fe790..9cf3679 100644
--- a/public/css/style.css
+++ b/public/css/style.css
@@ -126,6 +126,7 @@ label {
 /*Loader*/

 .loader-overlay { background-color: rgba(1, 1, 1, 0.7); bottom: 0; left: 0; position: fixed; right: 0; top: 0; display: none; }
+.loader-overlay.loader-overlay-show { display: block; }
 .loader { position: absolute; left: 50%; top: 50%; border: 16px solid #f3f3f3; border-top: 16px solid #3498db; border-radius: 50%; width: 120px; height: 120px; animation: spin 2s linear infinite; margin: -60px 0 0 -60px; }
 @keyframes spin { 0% { transform: rotate(0deg); } 100% { transform: rotate(360deg); } }

diff --git a/public/images/logo_medium_res.png b/public/images/logo_medium_res.png
new file mode 100644
index 0000000..9e6010a
Binary files /dev/null and b/public/images/logo_medium_res.png differ
diff --git a/public/js/zxcvbn-async.js b/public/js/zxcvbn-async.js
new file mode 100644
index 0000000..482bdfb
--- /dev/null
+++ b/public/js/zxcvbn-async.js
@@ -0,0 +1,21 @@
+(function() {
+
+  var ZXCVBN_SRC = 'https://cdnjs.cloudflare.com/ajax/libs/zxcvbn/4.2.0/zxcvbn.js';
+
+  var async_load = function() {
+    var first, s;
+    s = document.createElement('script');
+    s.src = ZXCVBN_SRC;
+    s.type = 'text/javascript';
+    s.async = true;
+    first = document.getElementsByTagName('script')[0];
+    return first.parentNode.insertBefore(s, first);
+  };
+
+  if (window.attachEvent != null) {
+    window.attachEvent('onload', async_load);
+  } else {
+    window.addEventListener('load', async_load, false);
+  }
+
+}).call(this);
diff --git a/src/controllers/admin/users.ts b/src/controllers/admin/users.ts
index c54bb4f..abdd8da 100644
--- a/src/controllers/admin/users.ts
+++ b/src/controllers/admin/users.ts
@@ -17,7 +17,7 @@ router.post("/signin", passport.authenticate("local", { failWithError: true }),
         return response.redirect("/admin");
     },
     function(err: any, request: any, response: any, next: any) {
-        (request.session as any).alertMessage = "Invalid username or passwaord";
+        (request.session as any).alertMessage = "Invalid username or password";
         return response.redirect("/admin/signin");
 });

diff --git a/src/views/components/HeaderImage.jsx b/src/views/components/HeaderImage.jsx
new file mode 100644
index 0000000..3a408b7
--- /dev/null
+++ b/src/views/components/HeaderImage.jsx
@@ -0,0 +1,12 @@
+import React, { Component } from 'react';
+
+export default class HeaderImage extends Component {
+    render() {
+        return (
+            <img className="text-center" alt="SHORE Centre"
+                 srcSet="/images/logo_medium_res.png,
+                         /images/logo_high_res.png 2x"
+                 src="/images/logo_high_res.png" />
+        );
+    }
+}
diff --git a/src/views/components/Loader.jsx b/src/views/components/Loader.jsx
index 265627e..13d21ac 100644
--- a/src/views/components/Loader.jsx
+++ b/src/views/components/Loader.jsx
@@ -4,7 +4,7 @@ class Loader extends Component {

     render() {
         return(
-            <div className="loader-overlay" id="loader-overlay">
+            <div className={`loader-overlay ${this.props.showByDefault ? 'loader-overlay-show' : ''}`} id="loader-overlay">
                 <div className="loader" id="loader"></div>
             </div>
         );
diff --git a/src/views/pages/Layout.jsx b/src/views/pages/Layout.jsx
index 3df0958..6bd263b 100644
--- a/src/views/pages/Layout.jsx
+++ b/src/views/pages/Layout.jsx
@@ -19,27 +19,24 @@ class Layout extends Component {
                     <title>{ this.props.title }</title>

                     <link rel="stylesheet" href="https://dhbhdrzi4tiry.cloudfront.net/cdn/sites/foundation.min.css"></link>
+                    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.11.4/themes/redmond/jquery-ui.min.css"></link>
                     <link rel="stylesheet" href="/css/style.css"></link>
-                    <link rel="stylesheet" href="https://code.jquery.com/ui/1.11.4/themes/redmond/jquery-ui.css"></link>
-            		<link rel="stylesheet" href="/js/libs/cleditor/jquery.cleditor.css"></link>
-
-                    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
-                    <script src="/js/global.js"></script>
-                    <script src="https://use.fontawesome.com/b3040caf6b.js"></script>
-                    <script src="https://code.jquery.com/jquery-1.10.2.js"></script>
-                    <script src="https://code.jquery.com/ui/1.10.4/jquery-ui.js"></script>
+                    <link rel="stylesheet" href="/js/libs/cleditor/jquery.cleditor.css"></link>

-                    <script src="https://cdnjs.cloudflare.com/ajax/libs/zxcvbn/4.2.0/zxcvbn.js"></script>
-            		<script src="/js/libs/cleditor/jquery.cleditor.min.js"></script>
-
-                    { scriptTags }
                     { styleTags }
-
                 </head>

                 <body>
-                {this.props.children}
-                <Loader />
+                    {this.props.children}
+                    <Loader showByDefault={this.props.showLoaderByDefault} />
+
+                    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
+                    <script src="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.11.4/jquery-ui.min.js"></script>
+                    <script src="https://use.fontawesome.com/b3040caf6b.js"></script>
+                    <script src="/js/global.js"></script>
+                    <script src="/js/libs/cleditor/jquery.cleditor.min.js"></script>
+
+                    { scriptTags }
                 </body>
             </html>
         );
diff --git a/src/views/pages/admin/index.jsx b/src/views/pages/admin/index.jsx
deleted file mode 100644
index bd3d4e4..0000000
--- a/src/views/pages/admin/index.jsx
+++ /dev/null
@@ -1,21 +0,0 @@
-import React, { Component } from "react";
-import Layout from "../Layout";
-
-class SignInPage extends Component {
-  render() {
-    return (
-        <Layout title="SHORE Admin Page">
-            <div id="shore_logo" className="row small-11 medium-11 large-11 columns text-center">
-                <img className="text-center" alt="SHORE Centre" src="images/logo.png" />
-            </div>
-
-            <div id="shore_page_0" className="shore-page-container">
-                <div className="row small-11 medium-8 large-5 columns text-center">
-                    <p className="shore-landing">Index Page</p>
-                </div>
-            </div>
-        </Layout>
-    )}
-}
-
-export default SignInPage
diff --git a/src/views/pages/admin/signin.jsx b/src/views/pages/admin/signin.jsx
index 6d6a7a5..cc57a0b 100644
--- a/src/views/pages/admin/signin.jsx
+++ b/src/views/pages/admin/signin.jsx
@@ -1,17 +1,21 @@
 import React, { Component } from "react";
 import Layout from "../Layout";
+import HeaderImage from "../../components/HeaderImage";
 import Messages from "../../components/Messages";
 import NavigationBar from "../../components/NavigationBar";

+const signInScripts = ["messages.js"];
+const signInStyles = ["zeitspace.css"];
+
 class SignInPage extends Component {
   render() {
     return (
-        <Layout title="SHORE Admin Page">
+        <Layout title="SHORE Admin Page" scripts={signInScripts} styles={signInStyles}>

             <Messages feedbackMessages={this.props.feedbackMessages} />

-            <div id="shore_logo" className="row small-11 medium-11 large-11 columns text-center">
-                <img className="text-center" alt="SHORE Centre" src="/images/logo.png" />
+            <div id="shore_logo" className="row small-11 medium-8 large-5 columns text-center">
+                <HeaderImage />
             </div>

             <div id="shore_page_0" className="shore-page-container">
@@ -39,10 +43,10 @@ class SignInPage extends Component {
                             <input  type="submit"
                                     value="Sign In!"
                                     name="submit"
-                                    className="button" />
+                                    className="button" />
                         </p>
                     </div>
-                </form>
+                </form>
             </div>
         </Layout>
     )}
diff --git a/src/views/pages/admin/users/createUser.jsx b/src/views/pages/admin/users/createUser.jsx
index 05498ed..4b798e3 100644
--- a/src/views/pages/admin/users/createUser.jsx
+++ b/src/views/pages/admin/users/createUser.jsx
@@ -4,7 +4,7 @@ import NavigationBar from "../../../components/NavigationBar";
 import Messages from "../../../components/Messages";
 import UserForm from "../../../components/UserForm";

-const scripts = ["messages.js", "securePassword.js"];
+const scripts = ["messages.js", "zxcvbn-async.js", "securePassword.js"];
 const styles = ["securePassword.css"];

 class CreateUserPage extends Component {
diff --git a/src/views/pages/admin/users/listUsers.jsx b/src/views/pages/admin/users/listUsers.jsx
index bcd7035..0162ff1 100644
--- a/src/views/pages/admin/users/listUsers.jsx
+++ b/src/views/pages/admin/users/listUsers.jsx
@@ -6,9 +6,10 @@ import UserTable from "../../../components/UserTable";
 import UserPasswordModal from "../../../components/UserPasswordModal";

 const usersStyles = ["usersList.css", "securePassword.css"];
-const usersScripts = ["usersList.js", "libs/foundation.min.js", "messages.js", "libs/jquery.form.min.js", "securePassword.js"];

-class SignInPage extends Component {
+const usersScripts = ["usersList.js", "libs/foundation.min.js", "messages.js", "libs/jquery.form.min.js", "zxcvbn-async.js", "securePassword.js"];
+
+export default class ListUsers extends Component {
   render() {
     return (
         <Layout title="SHORE Admin Page | Users" styles={usersStyles} scripts={usersScripts}>
@@ -27,5 +28,3 @@ class SignInPage extends Component {
         </Layout>
     )}
 }
-
-export default SignInPage
\ No newline at end of file
diff --git a/src/views/pages/index.jsx b/src/views/pages/index.jsx
index 1913f6e..9570e39 100644
--- a/src/views/pages/index.jsx
+++ b/src/views/pages/index.jsx
@@ -1,5 +1,6 @@
 import React, { Component } from "react";
 import Layout from "./Layout";
+import HeaderImage from "../components/HeaderImage";
 import Messages from "../components/Messages";

 const appScripts = ["surv3yjs/survey-engine.js", "shore-survey.js", "libs/moment.min.js", "messages.js", "libs/foundation.min.js", "libs/mixpanel.min.js"];
@@ -10,13 +11,13 @@ const usersScripts = ["usersList.js", "libs/foundation.min.js", "messages.js", "
 class MainPage extends Component {
   render() {
     return (
-        <Layout title="SHORE Referrals" scripts={appScripts} styles={appStyles}>
+        <Layout title="SHORE Referrals" scripts={appScripts} styles={appStyles} showLoaderByDefault={true}>
             <Messages feedbackMessages={this.props.feedbackMessages} />

             <div id="shore_logo" className="row">
                 <div className="small-12 medium-8 large-6 columns small-centered">
                     <a href="http://www.shorecentre.ca/" title="SHORE Center">
-                        <img className="text-center" alt="SHORE Centre" src="/images/logo_high_res.png" />
+                        <HeaderImage />
                     </a>
                 </div>
             </div>
